sql/codegen_alps.go

```diff
@@ -224,82 +223,79 @@ from alps.io.reader.odps_reader import OdpsReader
     def _build(self, experiment, run_config):
         return {{.ModelCreatorCode}}
 
-
 if __name__ == "__main__":
-
-	odpsConf=OdpsConf(
-		accessid="{{.OdpsConf.AccessID}}",
-		accesskey="{{.OdpsConf.AccessKey}}",
-		endpoint="{{.OdpsConf.Endpoint}}"
-	)
-	
-	trainDs = DatasetX(
-		num_epochs={{.TrainClause.Epoch}},
-		batch_size={{.TrainClause.BatchSize}},
-		shuffle="{{.TrainClause.EnableShuffle}}" == "true",
-		shuffle_buffer_size={{.TrainClause.ShuffleBufferSize}},
-{{if eq .TrainClause.EnableCache true}}
-		cache_file={{.TrainClause.CachePath}},
+    odpsConf=OdpsConf(
+        accessid="{{.OdpsConf.AccessID}}",
+        accesskey="{{.OdpsConf.AccessKey}}",
+        endpoint="{{.OdpsConf.Endpoint}}"
+    )
+
+    trainDs = DatasetX(
+        num_epochs={{.TrainClause.Epoch}},
+        batch_size={{.TrainClause.BatchSize}},
+        shuffle="{{.TrainClause.EnableShuffle}}" == "true",
+        shuffle_buffer_size={{.TrainClause.ShuffleBufferSize}},
+{{if .TrainClause.EnableCache}}
+        cache_file={{.TrainClause.CachePath}},
 {{end}}
-		reader=OdpsReader(
-			odps=odpsConf,
-			project="{{.OdpsConf.Project}}",
-			table="{{.TrainInputTable}}",
-			field_names={{.Fields}},
-			features={{.X}},
+        reader=OdpsReader(
+            odps=odpsConf,
+            project="{{.OdpsConf.Project}}",
+            table="{{.TrainInputTable}}",
+            field_names={{.Fields}},
+            features={{.X}},
 			labels={{.Y}},
 {{if ne .FeatureMapTable ""}}
-			feature_map=FeatureMap(table="{{.FeatureMapTable}}",
+            feature_map=FeatureMap(table="{{.FeatureMapTable}}",
 {{if ne .FeatureMapPartition ""}}
-				partition="{{.FeatureMapPartition}}"
+                partition="{{.FeatureMapPartition}}"
 {{end}}
-			)
+            )
 {{end}}
-		),
-		drop_remainder="{{.TrainClause.DropRemainder}}" == "true"
-	)
-
-	evalDs = DatasetX(
-		num_epochs=1,
-		batch_size={{.TrainClause.BatchSize}},
-		reader=OdpsReader(
-			odps=odpsConf,
-			project="{{.OdpsConf.Project}}",
-			table="{{.EvalInputTable}}",
-			field_names={{.Fields}},
-			features={{.X}},
-			labels={{.Y}}
-		)
-	)
-
-	export_path = "{{.ModelDir}}"
-
-	experiment = Experiment(
-		user="sqlflow",
-		engine=LocalEngine(),
-		train=TrainConf(input=trainDs,
-{{if ne .TrainClause.MaxSteps -1}}
-						max_steps={{.TrainClause.MaxSteps}},
+        ),
+        drop_remainder="{{.TrainClause.DropRemainder}}" == "true"
+    )
+
+    evalDs = DatasetX(
+        num_epochs=1,
+        batch_size={{.TrainClause.BatchSize}},
+        reader=OdpsReader(
+        odps=odpsConf,
+            project="{{.OdpsConf.Project}}",
+            table="{{.EvalInputTable}}",
+            field_names={{.Fields}},
+            features={{.X}},
+            labels={{.Y}}
+        )
+    )
+
+    export_path = "{{.ModelDir}}"
+
+    experiment = Experiment(
+        user="sqlflow",
+        engine=LocalEngine(),
+        train=TrainConf(input=trainDs,
+{{if (ne .TrainClause.MaxSteps -1)}}
```

Comment:
Just curious, is the parenthesis necessary of the if template?

Reply:
Yes, tested on my environment. Or the number `-1` will be recognized as an argument of the function call `.TrainClause.MaxSteps`